File: StriverA2Z/Java/BinaryTrees/RightView.java
import java.util.*;

public class RightView {
    public static void main(String[] args) {
        Node root = BinaryTreeOperations.arrayToBT(new Integer[] {1,2,3,4,5,6,7,8}, 0);
        List<Integer> ans = new ArrayList<>();
        rightView(root, 0, ans);
        for (Integer i: ans) {
            System.out.print(i + " ");
        }
        System.out.println();

    }

    public static void rightView(Node root, int level, List<Integer> ans) {
        if (root == null) return;
        if (level == ans.size()) ans.add(root.data);
        rightView(root.right, level + 1, ans);
        rightView(root.left, level + 1, ans);
    }

}

File: StriverA2Z/JavaScript&TypeScript/src/BinaryTrees/rightView.ts
import {arrayToBT, BTNode} from "./BinaryTreeOperations";

function rightView(root: BTNode, level: number, ans: number[]) {
	if (root === null) return;
	if (level === ans.length) ans.push(root.data);
	rightView(root.right, level + 1, ans);
	rightView(root.left, level + 1, ans);
}

let root = arrayToBT([1,2,3,4,5,6,7,8]);
let ans = [];
rightView(root, 0, ans);
console.log(ans)